### PR TITLE
chore(show-edit-pod): remove classic theme support - FE-3122

### PR DESCRIPTION
**Pull request by mkrds**
### Proposed behaviour
Remove classic theme support from ShowEditPod.
Rename `as` and `podType` to `variant` prop.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/show-edit-pod/show-edit-pod.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+          <Content title="Content title">
+            Content
+          </Content>
+        </ShowEditPod>
+        <div style={{ marginBottom: 24 }}/>
+      </>
+    )}
+  </Story>
+</Preview>
+
```
Comment (ianforster-sage) [inline src/components/show-edit-pod/show-edit-pod.stories.mdx:180]: Could we add demonstration of how to use a custom transition between view states?

File: cypress/features/regression/showEditPod.feature [hunk 2]
```diff
@@ -3,29 +3,29 @@ Feature: Show Edit Pod component
 
   @positive
   Scenario: Enable border checkbox for a Show Edit Pod component
-    When I open default "ShowEditPod" component in noIFrame with "showEditPod" json from "commonComponents" using "border" object name
+    When I open Default "ShowEditPod Test" component in noIFrame with "showEditPod" json from "commonComponents" using "border" object name
```
Comment (DlgSHi) [inline cypress/features/regression/showEditPod.feature:6]: This TC could be removed - you've added a `chromatic` example

File: cypress/features/regression/showEditPod.feature [hunk 3]
```diff
@@ -3,29 +3,29 @@ Feature: Show Edit Pod component
...
     Then Show Edit Pod component has border "rgb(204, 214, 219)" color
 
   @positive
   Scenario: Disable border checkbox for a Show Edit Pod component
-    When I open default "ShowEditPod" component in noIFrame with "showEditPod" json from "commonComponents" using "borderFalse" object name
+    When I open Default "ShowEditPod Test" component in noIFrame with "showEditPod" json from "commonComponents" using "borderFalse" object name
```
Comment (DlgSHi) [inline cypress/features/regression/showEditPod.feature:11]: This TC could be removed - you've added a `chromatic` example

File: cypress/features/regression/showEditPod.feature [hunk 4]
```diff
@@ -80,55 +80,55 @@ Feature: Show Edit Pod component
...
     Then Show Edit Pod title on preview is set to <title>
     Examples:
       | title                        | nameOfObject          |
       | mp150ú¿¡üßä                  | titleOtherLanguage    |
       | !@#$%^*()_+-=~[];:.,?{}&"'<> | titleSpecialCharacter |
 
   @positive
-  Scenario Outline: Set Show Edit Pod podType property to <podType>
-    When I open default "ShowEditPod" component in noIFrame with "showEditPod" json from "commonComponents" using "<nameOfObject>" object name
+  Scenario Outline: Set Show Edit Pod variant property to <variant>
```
Comment (DlgSHi) [inline cypress/features/regression/showEditPod.feature:103]: This TC could be removed - you've added a `chromatic` example

**Review by DlgSHi — COMMENTED** [review]
remove please `unnecessary` TC and `json` objects

**Review by ianforster-sage — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by ianforster-sage — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**